runtime-light/stdlib/job-worker/job-worker-api.cpp

```diff
@@ -0,0 +1,153 @@
+                                        .body = std::move(request)};
+  invoke_jw.store(tlb);
+
+  // send JW request
+  auto comp_query{co_await f$component_client_send_request(string{JOB_WORKER_COMPONENT_NAME}, string{tlb.data(), static_cast<string::size_type>(tlb.size())})};
+  if (comp_query.is_null()) {
+    php_warning("couldn't start job worker");
+    co_return INVALID_FORK_ID;
+  }
+  // create fork to wait for job worker response. we need to do it even if 'ignore_answer' is 'true' to make sure
+  // that the stream will not be closed too early. otherwise, platform may even not send job worker request
+  auto waiter_task{[](auto comp_query, std::chrono::nanoseconds timeout) noexcept -> task_t<string> {
+    auto fetch_task{f$component_client_fetch_response(std::move(comp_query))};
+    const string response{(co_await wait_with_timeout_t{task_t<string>::awaiter_t{std::addressof(fetch_task)}, timeout}).value_or(string{})};
+
+    tl::TLBuffer tlb{};
+    tlb.store_bytes(response.c_str(), static_cast<size_t>(response.size()));
+    tl::K2JobWorkerResponse jw_response{};
+    if (!jw_response.fetch(tlb)) {
+      co_return string{};
+    }
+    co_return std::move(jw_response.body);
+  }(std::move(comp_query), timeout_ns)};
+  // start waiter fork and return its ID
+  co_return(co_await start_fork_t{static_cast<task_t<void>>(std::move(waiter_task)), start_fork_t::execution::self});
+}
+
+} // namespace
+
+// ================================================================================================
+
+task_t<Optional<int64_t>> f$kphp_job_worker_start(string request, double timeout) noexcept {
+  const auto fork_id{co_await kphp_job_worker_start_impl(std::move(request), timeout, false)};
+  co_return fork_id != INVALID_FORK_ID ? fork_id : false;
+}
+
+task_t<bool> f$kphp_job_worker_start_no_reply(string request, double timeout) noexcept {
+  const auto fork_id{co_await kphp_job_worker_start_impl(std::move(request), timeout, true)};
+  co_return fork_id != INVALID_FORK_ID;
+}
+
+task_t<array<Optional<int64_t>>> f$kphp_job_worker_start_multi(array<string> requests, double timeout) noexcept {
+  array<Optional<int64_t>> fork_ids{requests.size()};
+  for (const auto &it : requests) {
+    const auto fork_id{co_await kphp_job_worker_start_impl(it.get_value(), timeout, false)};
+    fork_ids.set_value(it.get_key(), fork_id != INVALID_FORK_ID ? fork_id : false);
+  }
+  co_return fork_ids;
+}
+
+// ================================================================================================
+
+task_t<string> f$kphp_job_worker_fetch_request() noexcept {
+  if (!f$is_kphp_job_workers_enabled()) {
+    php_warning("couldn't fetch job worker request: job workers are disabled");
+    co_return string{};
+  }
+
+  auto &jw_server_ctx{JobWorkerServerComponentContext::get()};
+  if (jw_server_ctx.job_id == JOB_WORKER_INVALID_JOB_ID || jw_server_ctx.body.empty()) {
```

Comment:
Is it good decision to rely on the string move constructor? Maybe better to set `jw_server_ctx.job_id` to `JOB_WORKER_INVALID_JOB_ID` when runtime got body?

Reply:
Hmm... Good question. Actually, the first version of `job_worker_fetch_request` used the approach you mentioned. But I decided to change it to the current one. Why? I think that `job_id` should be valid until job worker is done, at the same time it seems quite logical that we can fetch the request only once, so I change `jw_server_ctx.body` to empty string